File: back/services/order/internal/model/order.go
package model

import (
	"SLGaming/back/pkg/snowflake"
	"time"

	"gorm.io/gorm"
)

// 订单状态常量
const (
	OrderStatusCreated   = 1 // 已创建，待支付/扣款
	OrderStatusPaid      = 2 // 已支付/已扣款，待陪玩接单
	OrderStatusAccepted  = 3 // 陪玩已接单，待开始服务
	OrderStatusInService = 4 // 服务中
	OrderStatusCompleted = 5 // 已完成
	OrderStatusCancelled = 6 // 已取消
	OrderStatusRated     = 7 // 已评价
)

// BaseModel 基础模型（与 user 服务风格保持一致）
type BaseModel struct {
	ID        uint64         `gorm:"primaryKey;autoIncrement:false" json:"id,string"`
	CreatedAt time.Time      `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deleted_at"`
}

// Order 订单表
type Order struct {
	BaseModel

	// 展示订单号（例如：202501010001），便于用户查看和客服查询
	OrderNo string `gorm:"size:32;uniqueIndex;not null;comment:订单号" json:"order_no"`

	// 老板 & 陪玩
	BossID      uint64 `gorm:"not null;index;comment:老板ID" json:"boss_id,string"`
	CompanionID uint64 `gorm:"not null;index;comment:陪玩ID" json:"companion_id,string"`

	// 游戏信息
	GameName string `gorm:"size:64;not null;comment:游戏名称" json:"game_name"`
	GameMode string `gorm:"size:64;comment:游戏模式/段位" json:"game_mode"`

	// 服务时长（分钟）
	DurationMinutes int32 `gorm:"not null;comment:服务时长(分钟)" json:"duration_minutes"`

	// 金额信息（帅币）
	PricePerHour int64 `gorm:"not null;default:0;comment:每小时价格(帅币)" json:"price_per_hour"`
	TotalAmount  int64 `gorm:"not null;default:0;comment:订单总价(帅币)" json:"total_amount"`

	// 状态
	Status int32 `gorm:"not null;index;comment:订单状态" json:"status"`

	// 时间字段
	PaidAt      *time.Time `gorm:"comment:支付时间" json:"paid_at"`
	AcceptedAt  *time.Time `gorm:"comment:接单时间" json:"accepted_at"`
	StartAt     *time.Time `gorm:"comment:开始服务时间" json:"start_at"`
	CompletedAt *time.Time `gorm:"comment:完成时间" json:"completed_at"`
	CancelledAt *time.Time `gorm:"comment:取消时间" json:"cancelled_at"`

	// 评价
	Rating  float64 `gorm:"type:decimal(3,2);not null;default:0;comment:评分(0-5)" json:"rating"`
	Comment string  `gorm:"type:text;comment:评价内容" json:"comment"`

	// 取消信息
	CancelReason string `gorm:"size:255;comment:取消原因" json:"cancel_reason"`
}

func (o *Order) TableName() string {
	return "orders"
}

// BeforeCreate 钩子：生成雪花主键
func (o *Order) BeforeCreate(tx *gorm.DB) error {
	if o.ID == 0 {
		o.ID = uint64(snowflake.GenID())
	}
	return nil
}

File: front/SLGaming/src/api/order/manage/order.js
import http from "@/utils/http";

/**
 * 获取订单详情
 *
 * @param {object} params - 查询参数
 * @param {number} [params.id] - 订单ID (int64) - 可选
 * @param {string} [params.orderNo] - 订单号 - 可选
 *
 * @returns {Promise}
 * @returns {object} response.data - OrderInfo
 * @returns {number} response.data.id - 订单ID
 * @returns {string} response.data.orderNo - 订单号
 * @returns {number} response.data.bossId - 老板ID
 * @returns {number} response.data.companionId - 陪玩ID
 * @returns {string} response.data.gameName - 游戏名称
 * @returns {number} response.data.durationHours - 时长（小时）
 * @returns {number} response.data.pricePerHour - 每小时价格（帅币）
 * @returns {number} response.data.totalAmount - 订单总价（帅币）
 * @returns {number} response.data.status - 状态：1=CREATED, 2=PAID, 3=ACCEPTED, 4=IN_SERVICE, 5=COMPLETED, 6=CANCELLED, 7=RATED
 * @returns {number} response.data.createdAt - 创建时间（时间戳）
 * @returns {number} response.data.paidAt - 支付时间（时间戳）
 * @returns {number} response.data.acceptedAt - 接单时间（时间戳）
 * @returns {number} response.data.startAt - 开始服务时间（时间戳）
 * @returns {number} response.data.completedAt - 完成时间（时间戳）
 * @returns {number} response.data.cancelledAt - 取消时间（时间戳）
 * @returns {number} response.data.rating - 评分（0-5）
 * @returns {string} response.data.comment - 评价内容
 * @returns {string} response.data.cancelReason - 取消原因
 */
export const getOrderDetailAPI = (params) => {
  return http({
    url: "/order",
    method: "GET",
    params,
  });
};

/**
 * 获取订单列表
 *
 * @param {object} params - 查询参数
 * @param {'boss' | 'companion'} [params.role] - 角色：boss / companion（默认 boss） - 可选
 * @param {number} [params.status] - 状态筛选：1=CREATED, 2=PAID, 3=ACCEPTED, 4=IN_SERVICE, 5=COMPLETED, 6=CANCELLED, 7=RATED - 可选
 * @param {number} [params.page] - 页码（从1开始） - 可选
 * @param {number} [params.pageSize] - 每页数量 - 可选
 *
 * @returns {Promise}
 * @returns {object} response.data - GetOrderListData
 * @returns {Array<object>} response.data.orders - 订单列表
 * @returns {number} response.data.total - 总数
 * @returns {number} response.data.page - 当前页码
 * @returns {number} response.data.pageSize - 每页数量
 */
export const getOrderListAPI = (params) => {
  return http({
    url: "/orders",
    method: "GET",
    params,
  });
};
